data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkConfig.java

```diff
@@ -36,10 +37,18 @@ public class S3SinkConfig {
     private AwsAuthenticationOptions awsAuthenticationOptions;
 
     @JsonProperty("bucket")
-    @NotEmpty
     @Size(min = 3, max = 500, message = "bucket length should be at least 3 characters")
     private String bucketName;
 
+    @JsonProperty("bucket_selector")
+    private PluginModel bucketSelector;
+
+    @AssertTrue(message = "Only one of bucket or bucket_selector option should be specified")
```

Comment:
```suggestion
    @AssertTrue(message = "You may not use both bucket and bucket_selector together in one S3 sink.")
```

This message should be more readable.